### DISCUSSION TITLE
TypeError when trying to create user account

### DISCUSSION THREAD

**Discussion by davidlontmx** [codeigniter4/shield#607] [Q&A] [answered]
I am receiving the following error when trying to create user account. This is on a new install, without any config modifications:

CodeIgniter\Shield\Controllers\RegisterController::getUserProvider(): Return value must be of type CodeIgniter\Shield\Models\UserModel, App\Models\UserModel returned 

Could anyone point me to a solution?

**Comment by kenjis** [comment, marked as answer]
Does `App\Models\UserModel` extend `CodeIgniter\Shield\Models\UserModel`?

**Reply by davidlontmx** [reply]
Thank you so very much. That was the problem. 

I had written: 

`class UserModel extends Model` 

But should have written 

`class UserModel extends \CodeIgniter\Shield\Models\UserModel`

Thanks @kenjis !

**Reply by davidlontmx** [reply]
Actually, I just realized something greater was wrong. I had created my own UserModel (for my own purposes), not realizing that there was already a UserModel as part of the Shield package.
